packages/gatsby/src/utils/page-data.ts

```diff
@@ -67,6 +69,54 @@ export function pageDataExists(publicDir: string, pagePath: string): boolean {
   return fs.existsSync(getFilePath(publicDir, pagePath))
 }
 
+let lmdbPageQueryResultsCache: GatsbyCacheLmdb
+function getLMDBPageQueryResultsCache(): GatsbyCacheLmdb {
+  if (!lmdbPageQueryResultsCache) {
+    const GatsbyCacheLmdbImpl = require(`./cache-lmdb`).default
+    lmdbPageQueryResultsCache = new GatsbyCacheLmdbImpl({
+      name: `internal-tmp-query-results`,
+    }).init()
+  }
+  return lmdbPageQueryResultsCache
+}
+
+export async function savePageQueryResult(
+  programDir: string,
+  pagePath: string,
+  result: IExecutionResult,
+  stringifiedResult: string
+): Promise<void> {
+  if (isLmdbStore()) {
+    await getLMDBPageQueryResultsCache().set(pagePath, result)
```

Comment:
I am not sure if we should `await` it here. LMDB-store is designed for batched transactions, `await`-ing can potentially make it slower, but worth benchmarking I guess.

Reply:
Also in my benchmark JSON-encoding results was also time-consuming. Would be great if we could switch to using query hash not based on results? Then we could save a bit more time here with `msgpackr` in lmdb.

Probably out of scope for this PR though.

Reply:
Dropped await when storing and also switched to storing stringified results (with string encoding on db) in https://github.com/gatsbyjs/gatsby/pull/32431/commits/fa84ff96e34b68982307a3e99c5c85d2948b510d